ls1intum/Artemis
# [bugfix] Show result in score -> submission page

### Checklist


### Motivation and Context



### Description

In the participations page of exam exercises, results of submissions which were not submitted (but had a result) were not displayed. 
This happend when a student hands in the exam (it therefore is submitted) but does not save the exercise, so there was no submission. This lead to a state in the database where the participation had the INITIALIZED state, but not the finished state.
![image](https://user-images.githubusercontent.com/33342534/114194500-db8f4d80-994f-11eb-96ab-68153b3614de.png)
This bugifx sets the InitializationsStates for all participations to FINISHED of the student exams which were submitted in time.

### Steps for Testing


1. create an exam with a modeling/text/file-upload exercise
2. participate with one student. Directly after starting the exam hand it in, without working on any exercise
3. participate with another student, submit all exercises but do not hand in the exam.
4. After the exam use the "assess unsubmitted student exams" in the student exam view
5. Assess all exams in the assessment dashboard
6. Check under exercise-groups -> participations if every result is visible


<details>
<summary>
Details
</summary>
Before the change:
<img src=https://user-images.githubusercontent.com/33342534/114195252-99b2d700-9950-11eb-861c-fd2b6d0e800d.png>
Note that user 1 submitted the exam, just did not save the exercise. (Directly after starting the exam he 
used hand in early)
User 50 did create a submission for the exercise, but did not hand in the exam.
After the change:
<img src=https://user-images.githubusercontent.com/33342534/114195071-6ff9b000-9950-11eb-95ba-c6961eb2a03b.png>

</details>
